sky/provision/__init__.py

```diff
@@ -43,6 +43,7 @@ def query_instances(
     provider_name: str,
     cluster_name: str,
     provider_config: Optional[Dict[str, Any]] = None,
+    non_terminated_only: bool = True,
```

Comment:
For discussion: Does it make sense to not expose this and always assume non_terminated_only=True? Will there be callers who would want this to be False?

stop() and terminate() for example already implicitly assume non-terminated, e.g.,
```
    filters = [
        {
            'Name': 'instance-state-name',
            # exclude 'shutting-down' or 'terminated' states
            'Values': ['pending', 'running', 'stopping', 'stopped'],
        },
        *_cluster_name_filter(cluster_name),
    ]
```
Also similar to node providers' design of get_nonterminated_nodes().

We can certainly leave this for the future.

Reply:
let me leave a comment about it